common/djangoapps/student/models.py

```diff
@@ -1340,6 +1344,31 @@ def is_verified_enrollment(self):
         """
         return CourseMode.is_verified_slug(self.mode)
 
+    @classmethod
+    def cache_key_name(cls, user_id, course_key):
+        """Return the name of the key to use to cache the current configuration
+        Args:
+            user_id(int): Id of user.
+            course_key(unicode): Unicode of course key
+
+        Returns:
+            Unicode cache key
+        """
+        return cls.COURSE_ENROLLMENT_CACHE_KEY.format(user_id, unicode(course_key))
+
+
+@receiver(models.signals.post_save, sender=CourseEnrollment)
+@receiver(models.signals.post_delete, sender=CourseEnrollment)
+def invalidate_enrollment_mode_cache(sender, **kwargs):  # pylint: disable=unused-argument
+    """Invalidate the cache of CourseEnrollment model. """
+    instance = kwargs['instance']
```

Comment:
Instead of getting the instance from `kwargs` I think you get the instance directly like the way implemented https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/models.py#L167